pkg/graveler/graveler.go

```diff
@@ -1684,31 +1712,41 @@ func (g *KVGraveler) Commit(ctx context.Context, repositoryID RepositoryID, bran
 		if params.SourceMetaRange != nil {
 			commit.MetaRangeID = *params.SourceMetaRange
 		} else {
-			changes, err := g.StagingManager.List(ctx, branch.StagingToken, ListingMaxBatchSize)
+			itrs, err := g.listSealedTokens(ctx, branch)
 			if err != nil {
-				return "", fmt.Errorf("staging list: %w", err)
+				return nil, err
+			}
+			if len(itrs) == 1 {
+				commit.MetaRangeID, _, err = g.CommittedManager.Commit(ctx, storageNamespace, branchMetaRangeID, itrs[0])
+			} else {
+				changes := NewCombinedIterator(itrs...)
+				defer changes.Close()
+				// returns err if the commit is empty (no changes)
+				commit.MetaRangeID, _, err = g.CommittedManager.Commit(ctx, storageNamespace, branchMetaRangeID, changes)
```

Comment:
1. can we consolidate - no need to have a specific case for one iterator the NewCombinedIterator should optimize it.
2. Looks like the call to Close on the listSealedTokens and combined iterator is not aligned